web/client-api/src/main/java/io/deephaven/web/client/api/storage/ItemDetails.java

```diff
@@ -0,0 +1,58 @@
+package io.deephaven.web.client.api.storage;
+
+import elemental2.core.JsArray;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.Storage_pb;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.storage_pb.FileInfo;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsProperty;
+
+public class ItemDetails {
+    private final String path;
+    private final int kind;
+    private final String size;
+
+    @JsMethod(namespace = "dh.storage.ItemDetails")
+    public static ItemDetails file(String... pathParts) {
+        return new ItemDetails(JsArray.asJsArray(pathParts).join("/"), Storage_pb.FileKind.getFILE(), null);
+    }
+
+    @JsMethod(namespace = "dh.storage.ItemDetails")
+    public static ItemDetails directory(String... pathParts) {
+        return new ItemDetails(JsArray.asJsArray(pathParts).join("/"), Storage_pb.FileKind.getDIRECTORY(), null);
+    }
+
+    public ItemDetails(String path, int kind, String size) {
+        this.path = path;
+        this.kind = kind;
+        this.size = size;
+    }
+
+    public static ItemDetails fromProto(FileInfo item) {
+        return new ItemDetails(item.getPath(), item.getKind(), item.getSize());
+    }
+
+    @JsProperty
+    public String getFilename() {
+        return path.substring(path.lastIndexOf('/'));
+    }
+
+    @JsProperty
+    public String getBasename() {
+        return path.substring(0, path.lastIndexOf('/'));
```

Comment:
This is incorrect - basename should be the just the last part of the path: https://en.wikipedia.org/wiki/Basename

I've included an example from the webdav impl below of what it returns. Some things to note that have come up elsewhere in this review:
- Filename is the full path
- Basename is just the last part of the path after the last slash (as noted on the wikipedia page linked above)
- No separate `path` property
- `type` for directory vs file
- It is not sorted, the sorting happens client side
- There's a spot for etag, but it's obviously `null` so not doing anything
- lastmod, mime, are interesting bits of detail, the Web UI does not use them though

```
[
  {
    "filename": "/bar",
    "basename": "bar",
    "lastmod": "Mon, 04 Oct 2021 22:43:08 GMT",
    "size": 0,
    "type": "directory",
    "etag": null
  },
  {
    "filename": "/images",
    "basename": "images",
    "lastmod": "Wed, 06 Oct 2021 19:33:51 GMT",
    "size": 0,
    "type": "directory",
    "etag": null
  },
  {
    "filename": "/linktest.md",
    "basename": "linktest.md",
    "lastmod": "Fri, 08 Apr 2022 13:01:59 GMT",
    "size": 192,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/ticking-size.py",
    "basename": "ticking-size.py",
    "lastmod": "Mon, 27 Jun 2022 14:19:22 GMT",
    "size": 1844,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/pete-examples",
    "basename": "pete-examples",
    "lastmod": "Mon, 04 Oct 2021 23:02:17 GMT",
    "size": 0,
    "type": "directory",
    "etag": null
  },
  {
    "filename": "/01 Basic Streams.md",
    "basename": "01 Basic Streams.md",
    "lastmod": "Thu, 07 Oct 2021 15:24:38 GMT",
    "size": 7135,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/introduction.md",
    "basename": "introduction.md",
    "lastmod": "Wed, 06 Oct 2021 21:04:08 GMT",
    "size": 3254,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/treemap",
    "basename": "treemap",
    "lastmod": "Fri, 09 Sep 2022 14:41:51 GMT",
    "size": 0,
    "type": "directory",
    "etag": null
  },
  {
    "filename": "/testtty.py",
    "basename": "testtty.py",
    "lastmod": "Wed, 13 Oct 2021 14:30:44 GMT",
    "size": 14,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/02 Stream and Batch Together.md",
    "basename": "02 Stream and Batch Together.md",
    "lastmod": "Fri, 29 Oct 2021 15:01:42 GMT",
    "size": 4981,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/03 Kafka Stream vs Append.md",
    "basename": "03 Kafka Stream vs Append.md",
    "lastmod": "Fri, 08 Oct 2021 19:23:48 GMT",
    "size": 6886,
    "type": "file",
    "etag": null,
    "mime": ""
  },
  {
    "filename": "/examples",
    "basename": "examples",
    "lastmod": "Fri, 10 Sep 2021 15:30:18 GMT",
    "size": 0,
    "type": "directory",
    "etag": null
  },
  {
    "filename": "/RandomValues.py",
    "basename": "RandomValues.py",
    "lastmod": "Fri, 02 Sep 2022 19:06:59 GMT",
    "size": 2212,
    "type": "file",
    "etag": null,
    "mime": ""
  }
]
```

Reply:
I think this is all resolved, but double check?